vault/plugin_catalog.go

```diff
@@ -157,6 +157,10 @@ func (c *PluginCatalog) cleanupExternalPlugin(name, id string) error {
 		return fmt.Errorf("plugin client not found")
 	}
 
+	if _, ok := extPlugin.connections[id]; !ok {
+		return fmt.Errorf("plugin connection not found")
```

Comment:
ID's are randomly generated and are really just used to as a unique ID for an instance of connection. I don't know if it would be useful to be honest.